<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

# Exercise 1

In [ ]:
while True:
    print('*', end='')

# The kernel was run and terminated by clicking the "Stop Cell Execution" button situated towards the left of the kernel.

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

# Exercise 2

In [6]:
print = 'PRINT'

In [ ]:
print('Hello World')

# The code was run and the kernel was restarted using the "Restart Kernel" button.

TypeError: 'str' object is not callable

# Exercise 3

In [ ]:
%%timeit

squares = [i**2 for i in range(10)]

661 ns ± 34.5 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [ ]:
%%timeit

squares = list()
for i in range(10):
    squares.append(i**2)

928 ns ± 89.7 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


Taking "661 ns ± 34.5 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)" as an example, 661 ns is the average time it takes to run the cell once and 34.5 ns is the standard deviation calculated from timing the code. A run represents a timing period, so 7 runs mean that 7 timings were taken. In each run, the cell was ran for 1,000,000 times (number of loops) and the timing of running the cell 1,000,000 times was recorded.